Source/JavaScriptCore/runtime/MachineContext.h

```diff
@@ -158,6 +158,14 @@ static inline void*& stackPointerImpl(mcontext_t& machineContext)
 {
 #if OS(DARWIN)
     return stackPointerImpl(machineContext->__ss);
+#elif OS(HAIKU)
+#if CPU(X86)
+    return reinterpret_cast<void*&>(machineContext.esp);
+#elif CPU(X86_64)
+    return reinterpret_cast<void*&>(machineContext.rsp);
+#else
+#error Unknown Architecture
+#endif
```

Comment:
Let's not have them if we do not need to support JIT.

Reply:
I'm not sure what you mean, as far as I know, JIT should be working at least on x86_64 for Haiku.
32-bit x86 seems not possible anymore in WebKit.
Other architectures will be added as Haiku becomes able to run on them well enough to support a WebKit browser.

Reply:
So yeah, X86 part should be removed